detectable/src/main/java/com/synopsys/integration/detectable/detectables/gradle/inspection/GradleInspectorExtractor.java

```diff
@@ -57,9 +57,21 @@ public Extraction extract(File directory, ExecutableTarget gradleExe, @Nullable
             gradleRunner.runGradleDependencies(directory, gradleExe, gradleInspector, gradleCommand, proxyInfo, outputDirectory);
 
             File rootProjectMetadataFile = fileFinder.findFile(outputDirectory, "rootProjectMetadata.txt");
-            List<File> reportFiles = fileFinder.findFiles(outputDirectory, "*_dependencyGraph.txt");
             List<CodeLocation> codeLocations = new ArrayList<>();
 
+            List<File> reportFiles = new ArrayList<>();
+
+            int depthCount = 0;
+            while(true) {
+                String pattern = "_depth" + depthCount;
+                List<File> tempList = fileFinder.findFiles(outputDirectory,"*"+pattern+"_dependencyGraph.txt");
+                if(tempList.isEmpty()) {
+                    break;
```

Comment:
Is it possible to have a gap of depths? Something like `0, 1, 3` (with `2` missing)?
If so, we may break out of the loop too early here.

Reply:
There is now a minuscule risk of an infinite loop: if for whatever reason there is a file named `{non-integer}_dependencyGraph.txt`, we will never reach the desired file count.
I understand that the code that generates these files does not do this now; however, I thought I'll mention it for your consideration.
What we could do here instead is sort the list of files by name and then process them sequentially.

Reply:
I did not want to do that because it would make the approach more complicated but I see your point so I am posting a commit implementing a Comparator that sorts by depth number.